osc-server/src/main/java/org/osc/core/broker/service/validator/ServiceFunctionChainDtoValidator.java

```diff
@@ -0,0 +1,130 @@
+import java.util.Set;
+
+import javax.persistence.EntityManager;
+
+import org.osc.core.broker.model.entities.virtualization.ServiceFunctionChain;
+import org.osc.core.broker.model.entities.virtualization.VirtualizationConnector;
+import org.osc.core.broker.model.plugin.ApiFactoryService;
+import org.osc.core.broker.service.dto.ServiceFunctionChainDto;
+import org.osc.core.broker.service.dto.VirtualSystemDto;
+import org.osc.core.broker.service.exceptions.VmidcBrokerValidationException;
+import org.osc.core.broker.service.persistence.OSCEntityManager;
+import org.osc.core.broker.service.persistence.VirtualizationConnectorEntityMgr;
+import org.osc.core.broker.util.TransactionalBroadcastUtil;
+import org.osc.core.broker.util.ValidateUtil;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+@Component(service = ServiceFunctionChainDtoValidator.class)
+public class ServiceFunctionChainDtoValidator implements DtoValidator<ServiceFunctionChainDto, ServiceFunctionChain> {
+	private EntityManager em;
+	private TransactionalBroadcastUtil txBroadcastUtil;
+
+	@Reference
+	ApiFactoryService apiFactoryService;
+
+	public ServiceFunctionChainDtoValidator create(EntityManager em) {
+		ServiceFunctionChainDtoValidator validator = new ServiceFunctionChainDtoValidator();
+		validator.em = em;
+		validator.apiFactoryService = this.apiFactoryService;
+		return validator;
+	}
+
+	@Override
+	public void validateForCreate(ServiceFunctionChainDto dto) throws Exception {
+		validate(dto);
+
+		OSCEntityManager<ServiceFunctionChain> emgr = new OSCEntityManager<ServiceFunctionChain>(
+				ServiceFunctionChain.class, this.em, this.txBroadcastUtil);
+
+		if (emgr.isExisting("name", dto.getName())) {
+			throw new VmidcBrokerValidationException("Service Function Name: " + dto.getName() + " already exists.");
+		}
+	}
+
+	@Override
+	public ServiceFunctionChain validateForUpdate(ServiceFunctionChainDto dto) throws Exception {
+		BaseDtoValidator.checkForNullId(dto);
+
+		ServiceFunctionChain sfc = this.em.find(ServiceFunctionChain.class, dto.getId());
+
+		if (sfc == null) {
+
+			throw new VmidcBrokerValidationException(
+					"Service Function Chain entry with name: " + dto.getName() + ") is not found.");
+		}
+
+		ValidateUtil.checkMarkedForDeletion(sfc, sfc.getName());
+
+		validate(dto, false);
+
+		return sfc;
+	}
+
+	void validate(ServiceFunctionChainDto dto) throws Exception {
+		validate(dto, true);
+	}
+
+	void validate(ServiceFunctionChainDto dto, boolean forCreate) throws Exception {
+		ServiceFunctionChainDtoValidator.checkForNullFields(dto);
+
+		if (!ValidateUtil.validateDaName(dto.getName())) {
+			throw new VmidcBrokerValidationException("Invalid Service Function Name: " + dto.getName()
+					+ "SFC name must not exceed 13 characters, must start with a letter, and can only contain numbers, letters and dash(-).");
+		}
+
+		Set<VirtualSystemDto> vsDtoList = dto.getVirtualizationSystems();
```

Comment:
couple of validations

1. Check every VS id is valid
2. Check that every VS -> Vc id matches the servicefunctionchain parent id